pulser-core/pulser/register/_coordinates.py

```diff
@@ -24,7 +27,7 @@ class CoordsCollection:
         _coords: The coordinates.
     """
 
-    _coords: np.ndarray | list
+    _coords: pm.AbstractArray | list
```

Comment:
If I am right, in Traps, we still define _coords from trap_coordinates, that is a `np.ndarray`. Either we go for 
```suggestion
    _coords: AbstractArrayLike | list
```
Or we transform trap_coordinates in traps.py `object.__setattr__(self, "_coords", pm.AbstractArray(trap_coordinates))`. By the way, do I get it right that trap_coordinates in Traps is not expected to be differentiable because it is related with a physical object and because it must be match some calibrated layouts that are not AbstractArray ?

Reply:
If I got right what you suggested above, won't `ArrayLike` work ?
```suggestion
    _coords: ArrayLike | list
```

Reply:
Here it's a bit tricky because it is both the type hint of both the input (which would be fine) but also the attribute itself - and there you run into issues.